### DISCUSSION TITLE
[9주차] 링크 계층 6.4~7

### DISCUSSION THREAD

**Discussion by Hanjaemo** [Dijkstra-Employ/Network-Study#34] [Q&A]
9주차 링크 계층 6.4~7에 해당하는 질의응답입니다.

**Comment by H-sooyeon** [comment]
## Q1
네트워크 계층의 IP 주소뿐만 아니라 링크 계층의 MAC가 추가로 필요한 이유에 대해 설명해주세요.

## Q2
ARP란 무엇이며, 어떻게 동작하는지 같은 서브넷일 경우와 같은 서브넷이 아닐 경우에 대해 나눠서 설명해주세요.

**Reply by Starlight258** [reply]
> ## Q1
> 네트워크 계층의 IP 주소뿐만 아니라 링크 계층의 MAC가 추가로 필요한 이유에 대해 설명해주세요.
> 
- IP 주소 : 네트워크 계층에서 목적지를 결정하여 라우팅하는데 사용된다. (라우팅, 호스트 식별)
- MAC 주소 : 장치를 식별하여 다음 홉을 결정하는데 사용된다. (직접 통신)

> ## Q2
> ARP란 무엇이며, 어떻게 동작하는지 같은 서브넷일 경우와 같은 서브넷이 아닐 경우에 대해 나눠서 설명해주세요.
- ARP : 동일한 서브넷 상에서 IP 주소를 MAC 주소로 변환한다.
- `같은 서브넷`일 경우 `ARP 테이블`을 찾아보고, 없을 경우 `ARP 패킷`을 브로드캐스팅한다.
- `다른 서브넷`일 경우 목적지의 서브넷에 도착할때까지 `라우팅 테이블`을 통해 `다음 홉 라우터의 MAC 주소`로 `목적지 MAC 주소`를 설정하여 이동시킨다.

**Comment by Starlight258** [comment]
# Q1
`MPLS`에 대해 설명하고, 사용하는 이유를 설명해주세요.

# Q2
데이터 센터 네트워킹 동향 중 하나를 골라 설명해주세요. (ex) `MDC` )

**Reply by H-sooyeon** [reply]
> Q1 MPLS에 대해 설명하고, 사용하는 이유를 설명해주세요.

MPLS는 네트워크의 라우터와 스위치 사이에서 데이터 패킷을 신속하게 전달하기 위해 사용된다.
MPLS는 네트워크의 각 패킷에 짧은 길이의 고유한 레이블을 부착한다.
라우터는 이 레이블을 기반으로 패킷의 목적지를 결정하고, 해당 패킷을 다음 네트워크 장치로 전달한다.
이를 통해 패킷을 전송할 때 IP 주소 대신 레이블만 확인하여 신속하게 경로를 결정할 수 있다.

MPLS를 사용하는 이유는 다음과 같다.
1. 빠른 데이터 전송 - IP 주소를 해석하는 과정을 줄여 네트워크를 더 빠르게 통과할 수 있다.
2. QoS 제공 - MPLS는 트래픽을 우선순위에 따라 처리할 수 있는 기능을 제공한다. 이로 인해 중요한 트래픽이 더 높은 우선순위로 처리될 수 있다.
3. 확장성 - 대규모 네트워크에서의 효율적인 트래픽 관리와 라우팅이 가능하며, 네트워크의 복잡성 증가에도 안정적인 성능을 유지할 수 있다.

<br />

> Q2 데이터 센터 네트워킹 동향 중 하나를 골라 설명해주세요. (ex) MDC )

SDN이 있다.
SDN은 네트워크의 제어 기능을 물리적인 네트워크 장비에서 분리하여, 소프트웨어를 통해 네트워크를 동적으로 관리하고 제어할 수 있게 하는 네트워크 아키텍처이다.
SDN은 중앙 제어로 관리되기에 유연하며, 데이터 센터 네트워크 확장시 복잡성을 줄이고 효율적으로 확장할 수 있는 기반을 제공한다.
또한 네트워크 트래픽을 중앙에서 모니터링하고 제어할 수 있어 잠재적인 보안 위협에 신속하게 대응할 수 있다.

**Reply by Starlight258** [reply]
> # Q1
> `MPLS`에 대해 설명하고, 사용하는 이유를 설명해주세요.
> 
- `다중 프로토콜 레이블 스위칭`으로 가상 회선 네트워크이다.
- `고정 길이 레이블`로 데이터그램을 전달하여, 전달 속도를 향상시킬 수 있다.
- `트래픽 엔지니어링`을 통해 여러 경로로 패킷을 전달할 수 있다.
- 링크 고장시 MPLS 전달 경로를 `복구`하거나, `VPN`을 구현하는데 사용된다. 

> # Q2
> 데이터 센터 네트워킹 동향 중 하나를 골라 설명해주세요. (ex) `MDC` )

`MDC` : 선반 컨테이너 기반의 모듈화된 데이터 센터
- 쉽게 배치 가능하고 확장 및 이동이 가능하다.
- 성능이 저하된 컨테이너부터 순차적으로 교체하거나 업그레이드할 수 있다.
- 임시 또는 긴급하게 컴퓨팅이 요구되거나 확장이 요구될 때 사용된다.

**Comment by choiyoung69** [comment]
## Q1
스위치의 필터링 및  포워딩이 어떻게 도착하는지 이해하기 위해서 목적지 주소 AA-AA-AA-AA-AA-AA를 가진 프레임이 스위치의 인터페이스 x에 도착했다고 가정합시다. 이때 일어날 수 있는 세 가지 경우에 대해 간단히 설명해주세요

## Q2
VLAN 트렁킹에 대해 간단히 설명해주세요
